### ISSUE TITLE
`statusCollection: Enabled` crashes controller when `statusType` is not defined

### ISSUE THREAD

**Issue by dionysius** [kubernetes-sigs/kubefed#1106]
**What happened**:
```bash
$ kubectl -n kube-federation-system logs -f kubefed-controller-manager-7f4858f959-cmmkq 
KubeFed controller-manager version: version.Info{Version:"v0.1.0-rc5", GitCommit:"99be0218bf5ac7d560ec0d7c2cfbfcbb86ba2d61", GitTreeState:"clean", BuildDate:"2019-08-01T16:50:07Z", GoVersion:"go1.12.5", Compiler:"gc", Platform:"linux/amd64"}
...
I0815 11:36:50.803433       1 controller.go:101] Starting sync controller for "FederatedService"
I0815 11:36:50.803499       1 controller.go:330] Started sync controller for "FederatedService"
E0815 11:36:50.883232       1 runtime.go:69] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
/go/src/sigs.k8s.io/kubefed/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:76
/go/src/sigs.k8s.io/kubefed/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:65
/go/src/sigs.k8s.io/kubefed/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:51
/usr/local/go/src/runtime/panic.go:522
/usr/local/go/src/runtime/panic.go:82
/usr/local/go/src/runtime/signal_unix.go:390
/go/src/sigs.k8s.io/kubefed/pkg/controller/status/controller.go:97
/go/src/sigs.k8s.io/kubefed/pkg/controller/status/controller.go:82
/go/src/sigs.k8s.io/kubefed/pkg/controller/federatedtypeconfig/controller.go:341
/go/src/sigs.k8s.io/kubefed/pkg/controller/federatedtypeconfig/controller.go:228
/go/src/sigs.k8s.io/kubefed/pkg/controller/util/worker.go:156
/go/src/sigs.k8s.io/kubefed/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133
/go/src/sigs.k8s.io/kubefed/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134
/go/src/sigs.k8s.io/kubefed/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88
/usr/local/go/src/runtime/asm_amd64.s:1337
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x50 pc=0x1547173]

goroutine 2488 [running]:
sigs.k8s.io/kubefed/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
        /go/src/sigs.k8s.io/kubefed/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:58 +0x105
panic(0x1794f60, 0x2d1e8e0)
        /usr/local/go/src/runtime/panic.go:522 +0x1b5
sigs.k8s.io/kubefed/pkg/controller/status.newKubeFedStatusController(0xc000340840, 0x1cdbbc0, 0xc0014c4400, 0xc000a639b8, 0x40bbd9, 0xc0015d7520)
        /go/src/sigs.k8s.io/kubefed/pkg/controller/status/controller.go:98 +0x93
sigs.k8s.io/kubefed/pkg/controller/status.StartKubeFedStatusController(0xc000340840, 0xc001608f00, 0x1cdbbc0, 0xc0014c4400, 0x0, 0x10)
        /go/src/sigs.k8s.io/kubefed/pkg/controller/status/controller.go:82 +0x5d
sigs.k8s.io/kubefed/pkg/controller/federatedtypeconfig.(*Controller).startStatusController(0xc000343c00, 0xc0011a72e0, 0xf, 0xc0014c5e00, 0x0, 0x0)
        /go/src/sigs.k8s.io/kubefed/pkg/controller/federatedtypeconfig/controller.go:341 +0xd7
sigs.k8s.io/kubefed/pkg/controller/federatedtypeconfig.(*Controller).reconcile(0xc000343c00, 0xc000e46280, 0x16, 0xc000e44600, 0x8, 0xc000e1b000)
        /go/src/sigs.k8s.io/kubefed/pkg/controller/federatedtypeconfig/controller.go:228 +0xae0
sigs.k8s.io/kubefed/pkg/controller/util.(*asyncWorker).worker(0xc0007274f0)
        /go/src/sigs.k8s.io/kubefed/pkg/controller/util/worker.go:156 +0xae
sigs.k8s.io/kubefed/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc000196e50)
        /go/src/sigs.k8s.io/kubefed/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x54
sigs.k8s.io/kubefed/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000196e50, 0x3b9aca00, 0x0, 0x1, 0xc0007a08a0)
        /go/src/sigs.k8s.io/kubefed/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134 +0xf8
sigs.k8s.io/kubefed/vendor/k8s.io/apimachinery/pkg/util/wait.Until(0xc000196e50, 0x3b9aca00, 0xc0007a08a0)
        /go/src/sigs.k8s.io/kubefed/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88 +0x4d
created by sigs.k8s.io/kubefed/pkg/controller/util.(*asyncWorker).Run
        /go/src/sigs.k8s.io/kubefed/pkg/controller/util/worker.go:119 +0xe5
```

**What you expected to happen**:
Should not crash the controller. Either validate error (e.g. writing to status of `federatedtypeconfigs service`) or use some well-defined defaults for `statusType` maybe?

**How to reproduce it (as minimally and precisely as possible)**:
1. Deploy kubefed
2. Edit `federatedtypeconfigs service` and add `statusCollection: Enabled`
3. See logs of controller

**Anything else we need to know?**:
Error happens here https://github.com/kubernetes-sigs/kubefed/blob/master/pkg/controller/status/controller.go#L97
And according to the CRD kubefedtypeconfigs:
>             statusType:
>               description: Configuration for the status type that holds information
>                 about which type holds the status of the federated resource. If not
>                 provided, the group and version will default to those provided for
>                 the federated type api resource.

**Environment**:
- Kubernetes version (use `kubectl version`) v1.15.1
- KubeFed version 0.1.0-rc5
- Scope of installation (namespaced or cluster) cluster probably

/kind bug

**Comment by marun** [comment]
This was fixed by #1070 and the fix will appear in the next release.

**Comment by marun** [comment]
Duplicate of #1069
